runtime/compiler/control/CompilationThread.cpp

```diff
@@ -4707,7 +4707,11 @@ bool TR::CompilationInfo::isQueuedForCompilation(J9Method * method, void *oldSta
    return linkageInfo->isBeingCompiled();
    }
 
+#if defined(OSX)
+JIT_HELPER(initialInvokeExactThunkGlue);
```

Comment:
Can the JIT_HELPER macro be updated to include `_` on OSX?  That would minimize the other required changes.

Reply:
The leading `_` is needed for platforms other than OSX. So what we would need is to have `JIT_HELPER(x)` defined to `extern "C" void x()` on mac and `extern "C" void _x()` on other platforms. We would then also need to enclose any calls to these functions in C/C++ in another macro, say, `CALL_JIT_HELPER`.

However, I did not go with that solution because the problem is that there is a lack of consistency here. The method calls/declarations with leading underscore in C also are named with leading underscores in the MASM code. Most of the methods are missing a leading underscore (and also in the assembly code), and so using our single `JIT_HELPER` macro that expands differently on OSX would not work. We would then need to divide them into 2 different categories, requiring 4 different macros: `JIT_HELPER(x)`, `JIT_HELPER_UNDERSCORE(x)`, `CALL_JIT_HELPER(x)`, `CALL_JIT_HELPER_UNDERSCORE(x)`.

Did you have a different idea in mind?

Reply:
The VM handles this somewhat differently by hiding the platform differences in the macros.  Looking at the JIT code, the approach there is to handle it with ifdefs throughout.

I don't want this to block this PR from moving forward.  Though I think it would be worth looking at cleaning this up in the future.

Reply:
This will be cleaned up once the NASM assembly files are set up to be used by Windows and xLinux. We will no longer need to have the code in the `#else` blocks of `#if defined(OSX)` in this PR.